VS_Meadow_Extension/VS_Meadow_Extension.Shared/MeadowConnection.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Meadow.CLI;
+using Meadow.Hcom;
+using System;
+using System.Threading;
+
+namespace Meadow
+{
+    static class MeadowConnection
+    {
+        static readonly int RETRY_COUNT = 10;
+        static readonly int RETRY_DELAY = 500;
+
+        internal static IMeadowConnection GetCurrentConnection()
```

Comment:
If possible it would be nice to have this static class centralised (in .Core or similar perhaps?) and shared between all the extensions, as they each require this functionality.